extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/ConnectionInterceptor.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public sealed interface ConnectionInterceptor permits TracesConnectionInterceptor,
+        ConnectionInterceptor.CompositeConnectionInterceptor {
+
+    void connectionOpened();
```

Comment:
Shouldn't we clarify what happens if an impl throws a runtime exception?

Reply:
I somehow don't expect that implementations will throw exception and it is sealed class, we would be explaining ourselves what happens if something that is not expected happens. Maybe I misunderstood you, but if you expect exceptions throw inside `connectionOpened`, I can handle them in `io.quarkus.websockets.next.runtime.telemetry.TelemetrySupport#connectionOpened` which is currently only place that really invokes it.